ipn/localapi/localapi_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package localapi
+
+import (
+	"runtime"
+	"testing"
+
+	"tailscale.com/client/tailscale/apitype"
+)
+
+func TestValidHost(t *testing.T) {
+	valid := []string{
+		"",
+		apitype.LocalAPIHost,
+		"localhost:41112", // port is the safesocket port on Windows
+		"127.0.0.1:41112",
+		"[::1]:41112",
+	}
+	if runtime.GOOS != "windows" {
+		valid = append(valid, "127.0.0.1:99")
+	}
+
+	for _, h := range valid {
+		if !validHost(h) {
+			t.Errorf("validHost(%q)=false, want true", h)
+		}
+	}
+
+	invalid := []string{
+		"100.100.100.100:41112",
+		"10.0.0.1:41112",
+		"37.16.9.210:41112",
+	}
```

Comment:
very JS style style test with two sections of good and bad :)

Reply:
I thought about defining types to parameterize the boolean but... is that really more readable? (I'm honestly not sure? I think not?) Parameterizing scales to more complex scenarios, but... tailscale.

Reply:
It's pretty idiomatic Go and we do it a 190 other places in the tree:

```
$ git grep -F '[]struct {' | wc -l
190
```

Reply:
+1 to table-driven

```go
for tc in ([]struct{host string, err error}{ .. }) {
  t.Run(tc.host, func(t *testing.T){ 
    ...
  })
}
```